client/www/js/controllers/routeManageController.js

```diff
@@ -0,0 +1,102 @@
+angular.module('amblr.routeManage', [])
+  .controller('routeManage', function ($scope, $rootScope, $ionicModal, $ionicSideMenuDelegate, $filter, Routes, POIs) {
+    $scope.disableSave = false;
+    $scope.userID = null;
+    $scope.availRoutes = [];
+    $scope.currentRoute = null;
+    $scope.routePOIs = [];
+    
+    $ionicModal.fromTemplateUrl('../../templates/routeManageModal.html', {
+      scope: $scope,
+      animation: 'slide-in-up',
+      // backdropClickToClose: true,
+      // hardwareBackButtonClose: true
+    })
+      .then(function (modal) {
+        $scope.modal = modal;
+      });
+
+    $scope.showRouteManageModal = function () {
+      Routes.getRoutes().then(function (routes) {
+        $scope.availRoutes = routes;
+        $scope.userID = $rootScope.userID;
+        $scope.modal.show();
+      });
+    }
+
+    $scope.hideRouteManageModal = function () {
+      $scope.modal.hide();
+      $scope.resetSelected();
+    }
+
+    $scope.selectRoute = function (routeID) {
+      var selectedRoute = $filter('filter')($scope.availRoutes, { '_id': routeID })[0];
+
+      /* filter and order by the route POIs order */      
+      $scope.routePOIs = POIs.inMemoryPOIs.reduce(function (accumulator, element, index) {
```

Comment:
Why inMemoryPOIs? Did we not have POIs available? Or is this so that we are only putting filtered POIs available in memory?